fix constraint truncation when too long

ruddell opened the pull request:
Fix #7537
Related to #7489

We used the same logic as when we shorten the table name, but only reduced the length by two when we add `_id` (length 3)

- Please make sure the below checklist is followed for Pull Requests.

- [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
- [x] Tests are added where necessary
- [X] Documentation is added/updated where necessary
- [X] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

GuiF007 commented:
If I'm not wrong, _.snakeCase('fooBar') becomes 'foo_bar', so you earn 1 character here ! Maybe it's better to truncate _.snakeCase('fooBar') ?

ruddell commented:
The variable is actually already in snake case and lower case from the [`getTableName` method](https://github.com/jhipster/generator-jhipster/blob/master/generators/generator-base.js#L1882-L1884) so I removed the extra calls to `_.snakeCase`.  Also added and improved the tests for this method

GuiF007 commented:
Ohh yeah, I saw that after my post, but this `_.snakeCase` disturbed me. Nice you removed it !
